pinot-core/src/test/java/org/apache/pinot/queries/MultiValueRawQueriesTest.java

```diff
@@ -807,4 +1320,283 @@ private void validateAggregateWithGroupByQueryResults(ResultTable resultTable, D
+        assertEquals(values.length, 11);
+        long count = (long) values[0];
+        long countRaw = (long) values[1];
+        assertEquals(count, 8);
+        assertEquals(count, countRaw);
+
+        double sum = (double) values[2];
+        double sumRaw = (double) values[3];
+        assertEquals(sum, sumRaw);
+        assertEquals(sum, expectedSumValues[i]);
+
+        double min = (double) values[4];
+        double minRaw = (double) values[5];
+        assertEquals(min, minRaw);
+        assertEquals(min, expectedMinValues[i]);
+
+        double max = (double) values[6];
+        double maxRaw = (double) values[7];
+        assertEquals(max, maxRaw);
+        assertEquals(max, expectedMaxValues[i]);
+
+        double avg = (double) values[8];
+        double avgRaw = (double) values[9];
+        assertEquals(avg, avgRaw);
+        assertEquals(avg, expectedAvgValues[i]);
+
+        assertEquals((int) values[10], (int) max);
+      }
+    }
+  }
+
+  private void validateAggregateWithGroupByOrderByQueryResults(ResultTable resultTable, DataSchema expectedDataSchema) {
+    assertNotNull(resultTable);
+    assertEquals(resultTable.getDataSchema(), expectedDataSchema);
+    List<Object[]> recordRows = resultTable.getRows();
+    assertEquals(recordRows.size(), 10);
+
+    double[] expectedSumValues = new double[]{400.0, 408.0, 416.0, 424.0, 432.0, 440.0, 448.0, 456.0, 464.0, 472.0};
+    double[] expectedMinValues = new double[]{0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0};
+    double[] expectedMaxValues = new double[]{100.0, 101.0, 102.0, 103.0, 104.0, 105.0, 106.0, 107.0, 108.0, 109.0};
+    double[] expectedAvgValues = new double[]{50.0, 51.0, 52.0, 53.0, 54.0, 55.0, 56.0, 57.0, 58.0, 59.0};
+
+    for (int i = 0; i < 10; i++) {
+      Object[] values = recordRows.get(i);
+      assertEquals(values.length, 11);
+      long count = (long) values[0];
+      long countRaw = (long) values[1];
+      assertEquals(count, 8);
+      assertEquals(count, countRaw);
+
+      double sum = (double) values[2];
+      double sumRaw = (double) values[3];
+      assertEquals(sum, sumRaw);
+      assertEquals(sum, expectedSumValues[i]);
+
+      double min = (double) values[4];
+      double minRaw = (double) values[5];
+      assertEquals(min, minRaw);
+      assertEquals(min, expectedMinValues[i]);
+
+      double max = (double) values[6];
+      double maxRaw = (double) values[7];
+      assertEquals(max, maxRaw);
+      assertEquals(max, expectedMaxValues[i]);
+
+      double avg = (double) values[8];
+      double avgRaw = (double) values[9];
+      assertEquals(avg, avgRaw);
+      assertEquals(avg, expectedAvgValues[i]);
+
+      if (expectedDataSchema.getColumnDataType(10) == DataSchema.ColumnDataType.LONG) {
+        assertEquals((long) values[10], i);
+      } else {
+        assertEquals((int) values[10], i);
+      }
+    }
+  }
+
+  @Test
+  public void testTransformInsideAggregateQueries() {
```

Comment:
Just saw this. Please ignore my earlier comment on nesting transform inside aggregate. It seems covered